terraform/environments/digital-prison-reporting/data_product_definitions.tf

```diff
@@ -0,0 +1,132 @@
+locals {
+  table_name = "${local.project}-data-product-definition"
+}
+
+module "dynamo_table_dpd" {
+  source              = "./modules/dynamo_tables"
+  create_table        = true
+  autoscaling_enabled = false
+  name                = local.table_name
+
+  hash_key    = "data-product-id"
+  range_key   = "category"
+  table_class = "STANDARD"
+  ttl_enabled = false
+
+  attributes = [
+    {
+      name = "data-product-id"
+      type = "S"
+    },
+    {
+      name = "category"
+      type = "S"
+    }
+  ]
+
+  global_secondary_indexes = [
+    {
+      name            = "category-index"
+      hash_key        = "category"
+      projection_type = "ALL"
+    }
+  ]
+
+  tags = merge(
+    local.all_tags,
+    {
+      Name          = local.table_name
+      Resource_Type = "Dynamo Table"
+    }
+  )
+}
+
+// Allow GitHub Actions run from the definitions repo to deploy/undeploy DPDs in the DynamoDB table.
+
+data "aws_iam_policy_document" "dpd_table_github_deploy_assume_role_policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+
+    principals {
+      type        = "Federated"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/token.actions.githubusercontent.com"]
+    }
+    condition {
+      test     = "StringEquals"
+      values   = ["sts.amazonaws.com"]
+      variable = "token.actions.githubusercontent.com:aud"
+    }
+    condition {
+      test     = "StringEquals"
```

Comment:
Can StringEquals compare correctly the value which contains a wildcard character?

Reply:
I updated this from Like to Equals based on the Checkov failure, which also included a wildcard. So I assumed it would work.

However, looking at the diocs, it seems it will not: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html#Conditions_String

Reply:
Fixed